glTF-Toolkit/src/GLTFMeshUtils.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include <numeric>
+#include <regex>
+#include <experimental\filesystem>
+
+#include <GLTFSDK/GLTF.h>
+#include <GLTFSDK/GLTFResourceWriter2.h>
+
+#include "GLTFMeshSerializationHelpers.h"
+#include "GLTFMeshUtils.h"
+
+using namespace Microsoft::glTF;
+using namespace Microsoft::glTF::Toolkit;
+using namespace std::experimental::filesystem;
+
+namespace
+{
+    class BasicStreamWriter : public IStreamWriter
+    {
+    public:
+        BasicStreamWriter(const std::string& OutputDirectory)
+            : m_OutputDir(OutputDirectory)
+        { }
+
+        std::shared_ptr<std::ostream> GetOutputStream(const std::string& Filename) const override
+        {
+            return std::make_shared<std::ofstream>(m_OutputDir + Filename, std::ios::binary);
+        }
+
+    private:
+        std::string m_OutputDir;
+    };
+}
+
+//-----------------------------------------
+// Main Entrypoint
+
+GLTFDocument GLTFMeshUtils::ProcessMeshes(const std::string& filename, const GLTFDocument& doc, const IStreamReader& reader, const MeshOptions& options, const std::string& outputDirectory)
```

Comment:
I believe this should take a StreamWriter instead of an output directory, to support other platforms